### ISSUE TITLE
Don't send emails to people who have already confirmed

### ISSUE THREAD

**Issue by KreativeKrise** [content-optimizer-gmbh/mautic-doi-plugin#5]
First of all, thank you for a great plugin!

But I have a use case. I'm not sure how to handle it with your plugin.
I offer some forms where people can get freebies. On the first request, the user should get the DOI email, which works fine. But when the same user requests another freebie, the user should not receive the DOI email again, but be sent directly to the "doi success segment".

Is there any way to archive this with your plugin?

**Comment by putzwasser** [comment]
Just an idea: Why not just use dynamic content?

If it is an unknown lead, show the form. If it is a known subscriber, show a download link, immediately.

**Comment by KreativeKrise** [comment]
For some, this is definitely a solution. In my case unfortunately not, because some forms ask for other fields.

The contacts are assigned to a main segment "marketing" and depending on which form they filled out or which data they entered, they land in a sub-segment, e.g. "industry X" or "interest Y".

**Comment by sebastian-fahrenkrog** [comment]
@KreativeKrise @putzwasser Thank you for your input! 

We use dynamic content in the past also. 

At the moment we got another mautic plugin ( JotaworksFormRulesBundle )  which allows to use a symfony expression language  to prevent form actions to fire again: 

For example we would do in the doi email action: 

{% FORM_RULES %}
lead.optin_status!="Bestätigt" and lead.optin_status!="Gestartet"
 {% END_FORM_RULES %}

With a form action like this: 

<img width="603" alt="Bildschirmfoto 2023-06-06 um 19 50 38" src="https://github.com/content-optimizer-gmbh/mautic-doi-plugin/assets/50595160/b593c409-4170-41c9-abd2-0143d75d6ad4">

But I will definitily checkout your merge request @KreativeKrise. Sounds amazing!

**Comment by putzwasser** [comment]
> For some, this is definitely a solution. In my case unfortunately not, because some forms ask for other fields.
>
>The contacts are assigned to a main segment "marketing" and depending on which form they filled out or which data they entered, they land in a sub-segment, e.g. "industry X" or "interest Y".

I still feel like this could be solved in another—maybe "better(?)" way.

Without having it tested:
- Create a DWC
- It shows either the form (no contact)
- Or a link to the freebie (identified contact)
- The link points to a mautic LP
- Use the Mautic JS tracking SDK to set the custom contact fields or tags or whatever you need, so that the contact gets added to the more specific segment.
  - For binary choices it'll be sufficient. Especially if downloading a freebie is connected to one specific segment. E.g., downloading freebie A ==> Contact comes into segment X. You can even filter for asset downloads in your segments:
  ![image](https://github.com/content-optimizer-gmbh/mautic-doi-plugin/assets/26040044/c869a43b-02ed-4848-be35-1fd486f8b1ef)
  - If this is not enough, because the contact needs so submit data (e.g., select an option or type text) you could create a form that asks for just that piece of data. Show it either on the landing page, before letting the user download the freebie, or link the form to your campaign/segment.
    - Either way: It is more comfortable for the user: They don't need to enter their data twice or more often.

(Full disclosure: I really don't like having (people) to sign up multiple times for a newsletter to get different freebies. I feel like that is bad user experience.)

I didn't review your PR, but: What if a contact is on DNC? Could they re-confirm their subscription?

**Comment by sebastian-fahrenkrog** [comment]
@putzwasser Thank you for your feedback and configuration! We use that pattern to and I agree: It's always good to not force multiple sign ups! 

But we wanted that feature for a little bit different reason: we don't wanted to allow leads to send multiple dnc emails to the same email address again and again 

>I didn't review your PR, but: What if a contact is on DNC? Could they re-confirm their subscription?

Yes, thats a good point. I think they could, if you make sure that they loose the segment/tag/lead field status after unsubscription. 

Some words about the normal plugin behaviour in case of do not contact: (in case anybody is wondering)

The normal mautic behaviour is: Do not contact status is NOT respected in form email actions. So no chance to block / blacklist an email. 

This could be a problem if the lead is send a legal warning to not send any more emails OR if there is a form spammer sending a dos attack to ONE email (and ruining your email reputation with that. of course with several email addresses he could still ruin your reputation. So you have to rate limit ip's etc. for that case. We use a custom Crowdsec Module to handle this case ) 

At the moment I included a check in the plugin for the current do not contact lead status. 

The doi email is only send IF the user set do not contact by his own action (and not set by mautic backend or by bounce handler) 

You can see the current logic in the code: 

https://github.com/content-optimizer-gmbh/mautic-doi-plugin/blob/9f7c2b9403a338222cda0df6caa8134a29037081/src/JotaworksDoiBundle/EventListener/FormSubscriber.php#L113

( BTW: I have to make sure that KreativeKrise pull request is still implementing this kind of behaviour. )

**Comment by KreativeKrise** [comment]
@sebastian-fahrenkrog 

> At the moment we got another mautic plugin ( JotaworksFormRulesBundle ) which allows to use a symfony expression language to prevent form actions to fire again

That's cool! If I had known that, the PR probably wouldn't have been necessary :sweat_smile: But maybe you find is usefull anyway.

> I have to make sure that KreativeKrise pull request is still implementing this kind of behaviour.

It's still implemented 👌 

@putzwasser

> I really don't like having (people) to sign up multiple times for a newsletter to get different freebies. I feel like that is bad user experience.

You are totally right! In the meantime, I have also changed that. We use the CMS TYPO3 and there is an extension with which it is possible to query whether a user is in a certain segment. If the user is now in the segment "Freebie", then I show him directly a button to download, otherwise the form. Therefore, my PR is currently no longer needed by us.

**Comment by sebastian-fahrenkrog** [comment]
>That's cool! If I had known that, the PR probably wouldn't have been necessary 😅 But maybe you find is usefull anyway

Absolutely! You can prevent that specific emails are getting spammed multiple times for example or if they bounce that you will ruin your mailserver reputation by sending them emails on and an! 

So really usefull! Thank you!

**Comment by putzwasser** [comment]
> We use the CMS TYPO3 and there is an extension with which it is possible to query whether a user is in a certain segment

That's neat and handy :+1: Could you share the link to it?
